eventmesh-common/src/main/java/org/apache/eventmesh/common/ThreadWrapper.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.eventmesh.common;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public abstract class ThreadWrapper implements Runnable {
+
+    private final AtomicBoolean started = new AtomicBoolean(false);
+    protected Thread thread;
```

Comment:
Besides its usage in `SubStreamHandler`, could `ThreadWrapper` be utilized in other classes as well?

`ThreadWrapper` appears to streamline synchronization for `SubStreamHandler`, eliminating the need for using a `latch`. However, the synchronization mechanism using `latch` in `SubStreamHandler` itself isn't overly complex. Hence, I'm curious whether introducing the additional complexity of `ThreadWrapper` for synchronization would address additional problems.